### PR TITLE
Cs 43

### LINKED ISSUE #43: Update AST format
The tags property needs to be constrained and split into some more specific properties.

### PR DESCRIPTION
**Pull request by thejoelw**
Fixes #43.

### CODE REVIEW

**Comment by Hopding** [comment]
@JWalker1995 Can you please link this PR to an issue, and explain what this is doing?

**Comment by thebho** [comment]
Still looking for a description on this PR

**Comment by Hopding** [comment]
Cross posting what Joel posted on #43:
> Changes to the AST node format:
> 1. The antlr node type is now stored in the `ast_type` property.
> 2. The `tags` property is now an array of "things" that the tag is. UI highlighting should depend solely on this property.
> 3. There is a `detail` property, which is an array of objects that contain extra details about the node. This is somewhat of a catch-all container, but will probably be used mostly to store documentation links and stuff like that. Each object in the array must have a `handler` property that specifies a function that handles the rest of the object.

File: public/index.html [hunk 1]
```diff
@@ -153,7 +154,7 @@
     let code_el = document.getElementById('code');
     let formatted_el = document.getElementById('formatted');
     code_el.onkeyup = code_el.onchange = function() {
-        let parser = module.exports;
+        let parser = window['Codesplain_parse_' + lang_el.value];
```
Comment (Hopding) [inline public/index.html]: This should be `window['CodesplainParser']`, per the change made in https://github.com/maryvilledev/codesplain/commit/c2ca09c1f5b9dc5702e351e04c41db872d598c8e.

**Review by Hopding — CHANGES_REQUESTED** [review]
Looks good, except for the one change I requested.

**Review by Hopding — APPROVED** [review]

**Pull request merged**